packages/core/src/App/Components/Elements/LiveChat/use-livechat.ts

```diff
@@ -0,0 +1,135 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useHistory } from 'react-router';
+import { liveChatInitialization } from './live-chat';
+import Cookies from 'js-cookie';
+import { deriv_urls } from '@deriv/shared';
+
+// Todo: Should move this declaration into a .d.ts file.
+declare global {
+    interface Window {
+        LiveChatWidget: {
+            on: (key: string, callback: VoidFunction) => void;
+            get: (key: string) => any;
+            call: (key: string, value?: any) => void;
+        };
+        LC_API: {
+            on_chat_ended: VoidFunction;
+        };
+    }
+}
+
+const useLiveChat = (has_cookie_account = false) => {
+    const [isReady, setIsReady] = useState(false);
+    const [reload, setReload] = useState(false);
+    const history = useHistory();
+
+    const liveChatDeletion = () =>
+        new Promise<void>(resolve => {
+            if (window.LiveChatWidget) {
+                window.LiveChatWidget.on('ready', () => {
+                    try {
+                        if (window.LiveChatWidget.get('customer_data').status !== 'chatting') {
+                            window.LiveChatWidget.call('destroy');
+                            resolve();
+                        }
+                    } catch (e) {
+                        resolve();
+                    }
+                });
+            } else {
+                resolve();
+            }
+        });
+
+    const onHistoryChange = useCallback(() => {
+        liveChatDeletion().then(() => {
+            liveChatInitialization().then(() => {
+                setReload(true);
+                setIsReady(true);
+            });
+        });
+    }, []);
+
+    const liveChatSetup = (is_logged_in: boolean) => {
+        window.LiveChatWidget?.on('ready', () => {
+            let client_first_name = '';
+            let client_last_name = '';
+            const domain = /^(.)*deriv\.(com|me)$/gi.test(window.location.hostname)
+                ? deriv_urls.DERIV_HOST_NAME
+                : 'binary.sx';
+            const client_information = Cookies.getJSON('client_information', {
+                domain,
+            });
+            const utm_data = Cookies.getJSON('utm_data', { domain });
```

Comment:
`getJSON` doesn’t exist on `Cookies`, Not sure how it is working now, Should We change this to `get` instead or keep it as it is? 🤔